### PR TITLE
drawer: support ESC to close it by on_action and call on_close

### LINKED ISSUE #263: Drawer: support ESC to close it

### PR DESCRIPTION
**Pull request by xda2023**
Fix #263

### CODE REVIEW

File: crates/ui/src/root.rs [hunk 1]
```diff
@@ -57,7 +57,14 @@ impl<'a> ContextModal for WindowContext<'a> {
             if root.active_drawer.is_none() {
                 root.previous_focus_handle = cx.focused();
             }
-            root.active_drawer = Some(Rc::new(build));
+
+            let focus_handle = cx.focus_handle();
+            focus_handle.focus(cx);
```
Comment (huacnlee) [inline crates/ui/src/root.rs:62]: We could't do `focus` here, this will make some unexpected bug.

Comment (huacnlee) [reply crates/ui/src/root.rs:62]: I was found this in Modal before, it's was also used `focus`, then I found some issue, so it now removed.

Ref: #250

Comment (huacnlee) [reply crates/ui/src/root.rs:62]: The notification layout not display now.

Comment (xda2023) [reply crates/ui/src/root.rs:62]: @huacnlee notify layer: The problem with the display not working can be solved by rebasing the changes onto the latest `main` branch. I suspect it has something to do with an update to a dependency version, possibly taffy

**Pull request merged**